### PR TITLE
Boilerplate remover.

### PR DESCRIPTION
**Pull request by berewt**

### CODE REVIEW

File: src/Singer/Stream.hs [hunk 1]
```diff
@@ -42,42 +43,44 @@ validTweet screenName t = not (retweeted t) && maybe True (== screenName) (inRep
...
+    t <- child'
+    ongoingThread t
+    child'
+  in verifiedChild <|> pure parent
+
+extractThread :: Maybe Tweet -> Maybe Thread
+extractThread tweet = do
+  t <- tweet
+  completedThread t
+  hasMultipleTweets t
```
Comment (adfretlink) [inline src/Singer/Stream.hs:68]: I'm not sure I understand this line, would you explain a bit what you're doing here ?

**Review by adfretlink — COMMENTED** [review]
Thank you for this much better rewrite !

- I have a question on a line where I'm not sure I get the idea.
- You formatted with stylish-haskell unless I'm mistaken ? If so, thanks for this too.

File: src/Singer/Stream.hs (hunk 1 above)
Comment (berewt) [reply src/Singer/Stream.hs:68]: I (vim actually) use britanny for formatting, so I basically did nothing but saving the files.
If you're talking about `extractThread`, the idea is to emit a thread if and only if we finished an actual thread, so we check if the thread is completed (no replyTo in the last tweet) and if it is actually a thread (it has replies), and we use the do-notation of the maybe monad for a clearer (maybe not so clearer, as you had to ask) syntax.

Comment (Raveline) [reply src/Singer/Stream.hs:68]: Ah - right, I think I have the idea - should child be nothing, the do block would stop here. Fair enough !

**Review by Raveline — APPROVED** [review]

File: src/Singer/Stream.hs (hunk 1 above)
Comment (berewt) [reply src/Singer/Stream.hs:68]: Yes, that's why there's a `void` there: I don't care about the child content, I just want to know that there's one. It ends up with a `Maybe ()` just like `guard`.

**Pull request merged**